src/views/Project/Details/Frameworks/FrameworkDetail/index.js

```diff
@@ -186,6 +245,27 @@ export default class FrameworkDetail extends React.PureComponent {
         });
     }
 
+    handleExportButtonClick = () => {
+        const {
+            requests: {
+                frameworkGetRequest: { response: framework },
+            },
+        } = this.props;
+
+        this.setState({
+            isExporting: true,
+        }, () => {
+            setTimeout(() => {
+                if (this.exportSectionRef.current) {
+                    html2canvas(this.exportSectionRef.current).then((canvas) => {
+                        saveAs(canvas.toDataURL(), `export-${framework.title}.png`);
+                        this.setState({ isExporting: false });
```

Comment:
This may be problematic as the async function will be called even after the component is unmounted.
We may need to look into how this can be safely done!